pymc3/distributions/continuous.py

```diff
@@ -1653,6 +1654,58 @@ def logcdf(self, value):
             tt.switch(tt.gt(y, 1), tt.log1p(-0.5 * tt.exp(-y)), tt.log(1 - 0.5 * tt.exp(-y))),
         )
 
+class AsymmetricLaplace(Continuous):
+    """
+    Assymetric Laplace Distribution
+    See also: https://en.wikipedia.org/wiki/Asymmetric_Laplace_distribution
+    """
+
+    def __init__(self, scale, symmetry, testval=0.0, *args, **kwargs):
+        """
+        Constructor
+        :param scale: scale parameter
+        :param symmetry: asymmetry parameter. Reduces to a normal laplace distribution with value 1
+        """
+        self.scale = tt.as_tensor_variable(scale)
+        self.symmetry = tt.as_tensor_variable(symmetry)
+
+        super().__init__(*args, **kwargs, testval=testval)
+
+    def random(self, point=None, size=None):
+        """
+        Draw random samples from this distribution, using the inverse CDF method.
+        :param point: not used
+        :param size: size of sample to draw
+        :return: Samples
+        """
+        if point is not None:
+            raise NotImplementedError('Random not implemented with point specified')
```

Comment:
If the parameters of a random variable are fixed by the point, then there is no need to draw it in model context. This is already implemented in [pm.distributions.distribution.draw_values](https://github.com/pymc-devs/pymc3/blob/master/pymc3/distributions/distribution.py#L680). So, I suspect these lines should not be there.

Reply:
Thanks @Sayam753 , I kinda already emitted that and It will reflect in the upcoming PR.